scripts/update_descriptions.py

```diff
@@ -0,0 +1,47 @@
+import requests
+import sys
+import hashlib
+import re
+import json
+import datetime
+
+def sha256(value):
+    m = hashlib.sha256()
+    m.update(value)
+    return m.hexdigest()
+
+if __name__ == '__main__':
+    if len(sys.argv) != 4:
+        print('Usage: ... <register> <old_description> <new_description>', file=sys.stderr)
+        sys.exit(1)
+    register = sys.argv[1]
+    old_desc = sys.argv[2]
+    new_desc = sys.argv[3]
+    print(f'<{register}> change "{old_desc}" -> "{new_desc}"', file=sys.stderr)
+
+    response = requests.get(f'https://{register}.beta.openregister.org/download-rsf')
+
+    old_item = None
+    old_hash = None
+    old_add_entry = None
+    for line in response.iter_lines():
+        match = re.match(rf'^add-item\t(.*{old_desc}.*)'.encode('utf8'), line)
```

Comment:
This works as long as we don't come across any other items with the same description in it. This is rare but sometimes we do have occasions where a register and an individual field will have the same description. This means that we risk picking up a `field:x` item here and accidentally combining it with a `register:x`entry.

Could you update the logic to scan for the correct item based on knowledge of its hash? So scan for the entry with key `register:x`, get the item-hash, then scan the items for the item with the corresponding hash. The item always occurs before the entry, which means either scanning the RSF file twice, or storing items and their item-hashes as you scan them. Not quite ideal for a little script but better than the risk of creating some dud RSF?

Reply:
Although using this approach you would have to make sure you pick up the latest entry with key `register:x` as it's possible for there to be multiple versions of it.

Reply:
Ok, I'll have a go at this. It's probably safer to just parse all the item JSONs and look up the description key that way.

Reply:
I had a go at implementing this yesterday afternoon. It's more complicated than the previous version but it's stricter. Is this what you had in mind? (see new commit)